libflux/flux-core/src/semantic/nodes.rs

```diff
@@ -319,7 +329,12 @@ impl File {
             imports.push(name);
 
             match importer.import(path) {
-                Some(poly) => env.add(name.to_owned(), poly),
+                Some(poly) => {
+                    // Types imported from other pacakges may have the same variables so we need
```

Comment:
My understanding is that this shouldn't be necessary because type instantiation will instantiate the variables. Do you think we can talk about this tomorrow so we can run through this together?

Reply:
It should. However any definition added here could contain variables that are later generated from the `Fresher`. If that happens then calling `Environment::apply` would be applied on these imported types as well which is wrong.

So say we import `f: forall a . a -> a` (a == Tvar(0), then the fresher also creates `Tvar(0)` and it gets unified with `int`. Once the substitution gets applied then the code as written will substitute `f: forall a . -> a => int -> int`  which is wrong because the `forall` (from the `PolyType`) should make these `Tvar(0)`s distinct.

I think it may be easier/safer to just fix the shadowing issue around `PolyType`s